pkg/graveler/retention/starting_point_iterator.go

```diff
@@ -0,0 +1,84 @@
+package retention
+
+import "github.com/treeverse/lakefs/pkg/graveler"
+
+// A GCStartingPoint represents a commit from which the GC algorithm should start scanning.
+// It could be either a branch HEAD, or a dangling commit.
```

Comment:
This confuses me about what the fields hold.  Specifically, does this mean that _either_ `BranchID` _or_ `CommitID` will be `nil`?  Or do I always get a `CommitID`?

Reply:
Added comment